pkg/model/event.go

```diff
@@ -235,30 +235,63 @@ func (e *Event) ContractName() string {
 	return e.contractName
 }
 
-// LogPayload returns "Raw" types.log field of event
-// NOTE(IS): by returning this, we are allowing the possibility of mutating the fields in this
+// LogPayload returns the log payload from the block
 func (e *Event) LogPayload() *types.Log {
-	return e.logPayload
+	// make a copy so fields are immutable
+	logPayloadCopy := &types.Log{Address: e.logPayload.Address, Topics: e.logPayload.Topics, Data: e.logPayload.Data, BlockNumber: e.logPayload.BlockNumber, TxHash: e.logPayload.TxHash, TxIndex: e.logPayload.TxIndex, BlockHash: e.logPayload.BlockHash, Index: e.logPayload.Index, Removed: e.logPayload.Removed}
```

Comment:
Should format this line so it isn't so long for legibility:
```
logPayloadCopy := &types.Log{
    Address: e.logPayload.Address,
    ...
}

```

Reply:
omg yes, sorry about that!!!